.github/workflows/verify-bundled-files.yml

```diff
@@ -0,0 +1,36 @@
+name: Verify Bundled Files
+
+on:
+  push:
+  pull_request:
+  schedule:
+   - cron: "0 1 * * *"
+  workflow_dispatch: ~
+
+permissions:
+  contents: read
+
+jobs:
+  VERIFY_BUNDLED_FILES:
+    name: Verify Bundled Files
+    runs-on: ubuntu-22.04
+    steps:
+      - name: git checkout
+        uses: actions/checkout@v5
+
+      - uses: dorny/paths-filter@v3
+        id: changes
+        with:
+          base: master
+          filters: |
+            pcre2:
+              - '.github/scripts/download-bundled/pcre2.sh'
+              - 'ext/pcre/pcre2lib/**'
+
+      - name: PCRE2 - Download
+        if: ${{ !cancelled() && (steps.changes.outputs.pcre2 == 'true' || (github.event_name != 'push' && github.event_name != 'pull_request')) }}
+        run: .github/scripts/download-bundled/pcre2.sh
+
+      - name: PCRE2 - Verify files
+        if: ${{ !cancelled() && (steps.changes.outputs.pcre2 == 'true' || (github.event_name != 'push' && github.event_name != 'pull_request')) }}
+        run: .github/scripts/test-directory-unchanged.sh ext/pcre/pcre2lib
```

Comment:
These can just be merged into a single step, possible with `::group::` markers for less YAML noise.